### DISCUSSION TITLE
Strongly Typed Connector IDs

### DISCUSSION THREAD

**Discussion by frankfka** [wevm/wagmi#3399] [Q&A]
Is there an exported type for, _or_ the possibility to derive, strongly typed connector IDs? We would like to attach certain metadata to different connectors - for example, we want to be able to distinguish a `metamask` injected connector from `walletConnect` or `okxwallet`. I currently don't see a way to make some ID -> metadata mapping that is typesafe.

**Comment by tmm** [comment]
Most stable "ID" on connectors is [`type`](https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/createConnector.ts#L37) (followed by `id`), but that doesn't enumerate all the possible injected wallets that could be added via [EIP-6963](https://wagmi.sh/react/api/createConfig#multiinjectedproviderdiscovery) (like `id` does).

**Reply by frankfka** [reply]
Is there an exported type for the `type` field?
